lib/rules/safe-context-keyword.js

```diff
@@ -50,16 +50,15 @@ module.exports.prototype = {
 
         // var that = this
         file.iterateNodesByType('VariableDeclaration', function(node) {
-            var firstToken = node.firstToken;
-
-            // Miss destructing assignment (#1699)
-            if (file.getNextToken(firstToken).value === '{') {
-                return;
-            }
 
             for (var i = 0; i < node.declarations.length; i++) {
                 var decl = node.declarations[i];
 
+                // Miss destructing assignment (#1699, #2119)
+                if (file.getNextToken(file.getPrevToken(decl)).value === '{') {
```

Comment:
+1. I like that, and I've confirmed it works. Would you like me to update my PR, or hold off due to your 3.0 point?